src/main/java/com/starbucks/service/OrderService.java

```diff
@@ -17,40 +18,41 @@ public OrderService(OrderRepository orderRepository) {
         this.orderRepository = orderRepository;
     }
 
-    public String saveOrder(Order order) {
-
-        orderRepository.save ( order );
-        return order.toString ( );
+    public Order saveOrder(Order order) {
+        return orderRepository.save(order);
     }
 
-    public String updateOrder(Order order , UUID id) {
-
-        if (orderRepository.existsById ( id )) {
-            Order orderDB = orderRepository.findById ( id ).get ( );
-            orderDB.setBeverages ( order.getBeverages ( ) );
-            orderDB.setSize ( order.getSize ( ) );
-            orderDB.setSugar ( order.getSugar ( ) );
-            orderRepository.save ( orderDB );
-        }
-        return "Order " + id + " not exist";
-
+    public Order updateOrder(UUID id, Order order) {
+        Order orderDB = findById(id);
+        orderDB.setBeverages(order.getBeverages());
+        orderDB.setSize(order.getSize());
+        orderDB.setSugar(order.getSugar());
+        return orderRepository.save(orderDB);
     }
 
     public boolean existOrder(UUID id) {
-        return orderRepository.existsById ( id );
+        try {
+            return orderRepository.existsById(id);
+        } catch (RuntimeException ex) {
+            throw new ResourceAccessException("Order does not exist");
+        }
     }
 
     public void deleteOrdersById(UUID id) {
-        orderRepository.deleteById ( id );
+        orderRepository.deleteById(id);
     }
 
-    public String patchOrder(UUID id , Status status) {
-        if (orderRepository.existsById ( id )) {
-            Order ordeDB = orderRepository.findById ( id ).get ( );
-            ordeDB.setStatus ( status );
-            orderRepository.save ( ordeDB );
-            return "Status changed!";
-        }
-        return "Id does not exist!";
+    public Order updateStatusOrder(UUID id, Status status) {
+        Order orderDB = findById(id);
```

Comment:
you should call an existing findById service method that will have the NOT FOUND check so you won't need to do it again.